src/codeu/chat/client/core/ConversationContext.java

```diff
@@ -114,5 +115,12 @@ private MessageContext getMessage(Uuid id) {
     final Iterator<Message> messages = view.getMessages(Arrays.asList(id)).iterator();
     return messages.hasNext() ? new MessageContext(messages.next(), view) : null;
   }
-
+  
+	public Iterable<String> listBots(){ 
+		final Collection<String> allBots = new ArrayList<>();
+	    for (String bot : view.getBots()) {
```

Comment:
You could have probably done this in one line, something like:

```java
final Collection<String> allBots = new ArrayList<>(view.getBots());
```

Or even:

```java
return new ArrayList<>(view.getBots());
```

Or maybe even:

```java
return view.getBots();
```

?